plugins/auth-backend/src/providers/aws-alb/provider.ts

```diff
@@ -100,34 +203,83 @@ export class AwsAlbAuthProvider implements AuthProviderRouteHandlers {
       return crypto.createPublicKey(optionalCacheKey);
     }
     const keyText: string = await fetch(
-      `https://public-keys.auth.elb.${this.options.region}.amazonaws.com/${keyId}`,
+      `https://public-keys.auth.elb.${this.region}.amazonaws.com/${keyId}`,
     ).then(response => response.text());
     const keyValue = crypto.createPublicKey(keyText);
     this.keyCache.set(keyId, keyValue.export({ format: 'pem', type: 'spki' }));
     return keyValue;
   }
 }
 
-export type AwsAlbProviderOptions = {};
+const awsAlbDefaultSignInResolver: SignInResolver<AwsAlbResult> = async (
+  info,
+  ctx,
+) => {
+  const { profile } = info;
+
+  if (!profile.email) {
+    throw new Error('Profile contained no email');
+  }
+
+  const userId = profile.email.split('@')[0];
+
+  const token = await ctx.tokenIssuer.issueToken({
+    claims: { sub: userId, ent: [`user:default/${userId}`] },
+  });
+
+  return { id: userId, token };
+};
+
+export type AwsAlbProviderOptions = {
+  /**
+   * The profile transformation function used to verify and convert the auth response
+   * into the profile that will be presented to the user.
+   */
+  authHandler?: AuthHandler<AwsAlbResult>;
 
-export const createAwsAlbProvider = (_options?: AwsAlbProviderOptions) => {
+  /**
+   * Configure sign-in for this provider, without it the provider can not be used to sign users in.
+   */
+  signIn?: {
+    /**
+     * Maps an auth result to a Backstage identity for the user.
+     */
+    resolver?: SignInResolver<AwsAlbResult>;
```

Comment:
To be honest I think we should go all the way with this migration and make the `resolver` here required, and not provide a default implementation at all. Should be fine here since the old experimental resolver was required as well.

Reply:
Yeah, sounds good. I will go for that then.